### PR TITLE
STENCIL-2598 - Add GeoTrust SSL Seal Toggle

### PR DESCRIPTION
**Pull request by mjschock**
## what (STENCIL-2598, formerly MERC-1160)
* adds a toggle for the GeoTrust SSL Seal
* implements the the instructions <a href="https://www.geotrust.com/support/seal/agreement/installation-instructions/">here</a>
* the domain and size can be set

@bigcommerce/stencil-team

### CODE REVIEW

File: templates/components/common/ssl-seal.html [hunk 1]
```diff
@@ -0,0 +1,8 @@
+<table width="135" border="0" cellpadding="2" cellspacing="0" title="Click to Verify - This site chose GeoTrust SSL for secure e-commerce and confidential communications.">
+    <tr>
+        <td width="135" align="center" valign="top">
+            <script type="text/javascript" src="https://seal.geotrust.com/getgeotrustsslseal?host_name={{theme_settings.geotrust_ssl_seal_hostname}}&amp;size={{theme_settings.geotrust_ssl_seal_size}}&amp;lang={{locale_name}}"></script><br />
+            <a href="http://www.geotrust.com/ssl/" target="_blank"  style="color:#000000; text-decoration:none; font:bold 7px verdana,sans-serif; letter-spacing:.5px; text-align:center; margin:0px; padding:0px;"></a>
+        </td>
+    </tr>
+</table>
```
Comment (willfaught) [inline templates/components/common/ssl-seal.html]: Missing NL

File: config.json [hunk 2]
```diff
@@ -590,4 +593,4 @@
       }
     }
   ]
-}
+}
```
Comment (willfaught) [inline config.json]: Missing NL

**Comment by willfaught** [comment]
🍹 Keep in mind https://github.com/bigcommerce/standards/blob/master/docs/Git.md. Can we add the ticket (MERC-1160?) to the summary and use a verb (add?) and sentence case (Add GeoTrust SSL...)?

File: templates/components/common/ssl-seal.html (hunk 1 above)
Comment (mjschock) [inline templates/components/common/ssl-seal.html]: hmmm...  yes, i see. the linter doesn't look at html or json.  wondering if we should add something to do the check.

Comment (mjschock) [inline templates/components/common/ssl-seal.html]: i've added a new line at the end but it doesn't seem to show up here, although you can see that there are now 9 lines:  https://github.com/mjschock/stencil/blob/f80cdfc06862a9668db449697261c8d9a07eb4a2/templates/components/common/ssl-seal.html

File: config.json (hunk 2 above)
Comment (mjschock) [inline config.json]: it's been added.  see https://github.com/mjschock/stencil/blob/f80cdfc06862a9668db449697261c8d9a07eb4a2/config.json.  
screen shot 2017-01-25 at 10 19 04 am

File: .eslintrc [hunk 3]
```diff
@@ -7,7 +7,7 @@
         "complexity": 1,
         "no-alert": 0,
         "consistent-return": 0,
-        "max-len": 0,
+        "max-len": 0
```
Comment (mcampa) [inline .eslintrc]: Unnecessary change. `.eslintrc` is not real json format 🍹 . 
You can have trailing commas and add comments in this file

Comment (mjschock) [inline .eslintrc]: yeah, just removed it because vscode doesn't like it.   it's been added back :)

File: config.json [hunk 4]
```diff
@@ -244,7 +244,10 @@
     "restrict_to_login": false,
     "swatch_option_size": "22x22",
     "social_icon_placement_top": false,
-    "social_icon_placement_bottom": "bottom_none"
+    "social_icon_placement_bottom": "bottom_none",
+    "show_geotrust_ssl_seal": false,
+    "geotrust_ssl_seal_hostname": "",
```
Comment (junedkazi) [inline config.json]: why not use `shopPath` instead ? Is there a reason for not using the store domain instead of offering a separate field ?

Comment (mcampa) [inline config.json]: Good question, however `shopPath` or `ShopPathSSL` return a url `http://...` not the hostname. But we could return the hostname in the `Setting` resource and use that

Comment (junedkazi) [inline config.json]: We do have `base_url` & `secure_base_url` for both http & https links. We also have the `home` in the `urls` section of the `context`. All the urls are with `http{s}://` so we will need to strip them out. The question is should we pass it as resource on the bcapp side or just add a helper in pencil to strip `http{s}://`.

Comment (mcampa) [inline config.json]: +1 It would be useful to have helper for that. Like `getHostname`

But the question is if merchants really need to enter their own domain

Comment (junedkazi) [inline config.json]: Since the trust ssl seal is for the store domain name. I don't see a reason why we should expose a field for the user to enter it unless there is some historic reason which I am not aware of.

**Comment by mcampa** [comment]
We are now using `CHANGELOG.md`. Please add an entry with your change under `draft` before merging

```
## Draft
- <here>

## 1.5.1 (2017-02-07)
- ...

```

File: CHANGELOG.md [hunk 5]
```diff
@@ -5,6 +5,7 @@
 - Prevent carousel images from being cut off on large screens by adding a new setting to theme editor schema [#909](https://github.com/bigcommerce/stencil/pull/909)
 - Add schema description specifying that social media icons must be set up to see them [#920](https://github.com/bigcommerce/stencil/pull/920)
 - Show account creation links only if it is enabled in store settings [#917] (https://github.com/bigcommerce/stencil/pull/917)
+- Add GeoTrust SSL Seal Toggle [#903] (https://github.com/bigcommerce/stencil/pull/903)
```
Comment (mjschock) [inline CHANGELOG.md]: @junedkazi - updated

**Comment by junedkazi** [comment]
@nickdengler can we test this one before we merge this one.

**Comment by nickdengler** [comment]
LGTM 💚
